### PR TITLE
dnu refactoring preparation

### PR DESCRIPTION
**Pull request by ChengTian**
- Fix NuGetFramework.ToString()
- Add workaround in LockFile to make generated lock file work with runtime
- Fix a bug in JsonPackageSpecReader

parent https://github.com/aspnet/dnx/issues/1623

@davidfowl @anurse @emgarten

### CODE REVIEW

File: src/NuGet.Frameworks/NuGetFramework.cs [hunk 1]
```diff
@@ -150,11 +150,11 @@ public string DotNetFrameworkName
                         parts.Add(String.Format(CultureInfo.InvariantCulture, "Profile={0}", Profile));
                     }
 
-                    result = String.Join(", ", parts);
+                    result = String.Join(",", parts);
```
Comment (ChengTian) [inline src/NuGet.Frameworks/NuGetFramework.cs]: @emgarten , I am removing the space after `,` to make it consistent with the framework name used by DNX runtime, which is using `System.Runtime.Versioning.FrameworkName`. Is it safe to make this change?

File: src/NuGet.ProjectModel/LockFile.cs [hunk 2]
```diff
@@ -61,5 +62,16 @@ public bool IsValidForPackageSpec(PackageSpec spec)
 
             return true;
         }
+
+        // DNU REFACTORING TODO: temp hack to make generated lockfile work with runtime lockfile validation
+        private static string RuntimeStyleLibraryRangeToString(LibraryRange libraryRange)
```
Comment (ChengTian) [inline src/NuGet.ProjectModel/LockFile.cs]: `NuGet.LibraryModel.LibraryRange.ToString()` returns something like
`PackageName [MinVersion,)`
while the `LibraryRange.ToString()` in runtime returns
`PackageName >= MinVersion-*`
The hack here makes them consistent

Comment (ChengTian) [inline src/NuGet.ProjectModel/LockFile.cs]: We need to change implementation of `LibraryRange.ToString()` in runtime but I want to make the PRs for refactoring minimal. That's why I introduce this hack.

File: src/NuGet.Frameworks/NuGetFramework.cs (hunk 1 above)
Comment (emgarten) [inline src/NuGet.Frameworks/NuGetFramework.cs]: Removing the space is fine. Just make sure all unit tests are updated to match the change.

Comment (ChengTian) [inline src/NuGet.Frameworks/NuGetFramework.cs]: Thanks for the reminder. I just updated unit tests.

**Comment by anurse** [comment]
:shipit:

**Pull request merged**